aikau/src/main/resources/alfresco/forms/controls/MultiSelect.js

```diff
@@ -1371,8 +1371,8 @@ define([
             this._focusedResult = null;
 
             // If we have too many current results displaying, remove the excess nodes
-            while (this._nodes.resultsDropdown.childNodes.length > this._results.length + 3) {
-               this._nodes.resultsDropdown.removeChild(this._nodes.resultsDropdown.lastChild);
+            while (this._nodes.resultsDropdown.childNodes.length > 3) {
```

Comment:
No. This is wrong. You should be removing excess results, not all results. The idea is to avoid re-generating DOM elements for every result if they're already there and can simply have their text content updated.